lib/stdlib/test/property_test/queue_prop.erl

```diff
@@ -636,22 +636,17 @@ do_op_list({Op, I}, L0)
        Op =:= cons->
     {undefined, [I|L0]};
 %% when the model list is empty empty, out/1, out_r/1, peek/1, peek_r/1,
-%% get/1, get_r/1, head/1, daeh/1 and last/1 return `empty' and do not
-%% change the model list
+%% get/1, get_r/1, head/1, daeh/1, init/1, last/1, drop/1, drop_r/1,
+%% tail/1 and liat/1 return `empty' and do not change the model list
 do_op_list(Op, [] = L0)
```

Comment:
This function still suffers from the semantics of operations being spread out and can be rewritten in a similar way as `do_op_queue`.  This both adds clarity and consistency.

Reply:
In the spirit of not being lazy, I include a suggested rewrite of `do_op_list/2` as well.
```

list_op_map() ->
    QueueOp1 = fun(F) -> fun(Q) -> {undefined, F(Q)} end end,
    QueueOp2 = fun(F) -> fun(I, Q) -> {undefined, F(I, Q)} end end,

    CatchEmpty = fun(F) ->
                         fun([]) -> {empty, []};
                            (L) -> F(L)
                         end
                 end,

    Wrap = fun(F) -> fun(L) -> {{value, F(L)}, L} end end,

    %% Add element to end of queue
    App = fun(I, L) -> L ++ [I] end,
    %% Add element at head of queue
    Cons = fun(I, L) -> [I|L] end,

    %% Remove and return first element of queue
    RemoveFirst = fun([H|T]) -> {{value, H}, T} end,
    %% Remove and return last element of queue
    RemoveLast = fun(L) -> {{value, lists:last(L)}, lists:droplast(L)} end,
    %% Get first element of queue, but do not remove it
    First = Wrap(fun erlang:hd/1),
    %% Get last element of queue, but do not remove it
    Last = Wrap(fun lists:last/1),

    %% Remove first element of queue without returning it
    DropFirst = QueueOp1(fun erlang:tl/1),
    %% Remove last element of queue without returning it
    DropLast = QueueOp1(fun lists:droplast/1),

    #{
      in => QueueOp2(App),
      snoc => QueueOp2(App),
      in_r => QueueOp2(Cons),
      cons => QueueOp2(Cons),

      out => CatchEmpty(RemoveFirst),
      out_r => CatchEmpty(RemoveLast),

      peek => CatchEmpty(First),
      get => CatchEmpty(First),
      head => CatchEmpty(First),

      peek_r => CatchEmpty(Last),
      get_r => CatchEmpty(Last),
      daeh => CatchEmpty(Last),
      last => CatchEmpty(Last),

      drop_r => CatchEmpty(DropLast),
      init => CatchEmpty(DropLast),
      liat => CatchEmpty(DropLast),

      drop => CatchEmpty(DropFirst),
      tail => CatchEmpty(DropFirst)
     }.

do_op_list({Op, I}, L) ->
    F = maps:get(Op, list_op_map()),
    F(I, L);
do_op_list(Op, L) ->
    F = maps:get(Op, list_op_map()),
    F(L).
```

Reply:
I changed the model functions to use an operations map, like for the queue functions, so they look pretty similar now.

Reply:
Oh, I noticed your suggestion only just now, ie after having it done by myself already :sweat_smile:  Thanks, though :hugs:

Reply:
I just took a closer look at your suggestion, and I'm amazed how much of it happens to be the same as in mine, especially `CatchEmpty`/`MaybeEmpty` 😄

Reply:
No surprise, really, considering the history of the conversation.